consumerBuilder.go

```diff
@@ -48,6 +49,11 @@ type (
 		saramaConfig        *sarama.Config
 		saramaClusterConfig *cluster.Config
 	}
+
+	topicConsumerBuilder struct {
```

Comment:
naming this Builder make the code very confusing. Looks like what you need is a ConsumerTopic along with a TopicType enum. One suggestion would be to create a package local version of ConsumerTopic struct that wraps the kafka.ConsumerTopic and adds a TopicType enum. If the TopicType is TopicTypeRetryQ or TopicTypeDLQ, then metadata can be automatically decoded by the partitionConsumer

Reply:
Done